File: src/main/java/com/github/joelws/release/tracker/service/release/execution/UpdateReleaseServiceExecution.java
package com.github.joelws.release.tracker.service.release.execution;

import com.github.joelws.release.tracker.dao.release.ReleaseDao;
import com.github.joelws.release.tracker.entity.release.Release;
import com.github.joelws.release.tracker.service.ServiceExecution;

public class UpdateReleaseServiceExecution implements ServiceExecution<Release, Release> {
    private final ReleaseDao releaseDao;

    public UpdateReleaseServiceExecution(ReleaseDao releaseDao) {
        this.releaseDao = releaseDao;
    }

    @Override
    public Release execute(Release param) {
        return (releaseDao.read(param.getName()) != null) ? releaseDao.update(param) : null;
    }
}

File: src/main/java/com/github/joelws/release/tracker/service/release/execution/UpdateReleaseServiceExecution.kt
package com.github.joelws.release.tracker.service.release.execution

import com.github.joelws.release.tracker.dao.release.ReleaseDao
import com.github.joelws.release.tracker.entity.release.Release
import com.github.joelws.release.tracker.service.ServiceExecution

open class UpdateReleaseServiceExecution(private val releaseDao: ReleaseDao) : ServiceExecution<Release, Release?> {

    override fun execute(param: Release): Release? {
        return if (releaseDao.read(param.name) != null) releaseDao.update(param) else null
    }
}
